src/Property.php

```diff
@@ -79,10 +82,23 @@ public function isNullable(): bool
 
     protected function resolveTypeDefinition()
     {
+        if (isset(self::$cache[$this->class][$this->getName()])) {
+            $this->isNullable = self::$cache[$this->class][$this->getName()]['is_nullable'];
+            $this->hasTypeDeclaration = (
+                ! empty(self::$cache[$this->class][$this->getName()]['types'])
+                || ! empty(self::$cache[$this->class][$this->getName()]['array_types'])
+            );
+            $this->types = self::$cache[$this->class][$this->getName()]['types'] ?? [];
+            $this->arrayTypes = self::$cache[$this->class][$this->getName()]['array_types'] ?? [];
+
+            return;
+        }
+
         $docComment = $this->getDocComment();
 
         if (! $docComment) {
             $this->isNullable = true;
+            self::$cache[$this->class][$this->getName()]['is_nullable'] = $this->isNullable;
```

Comment:
Why not store the class as a whole instead of its properties separately?

Reply:
I have an idea using clone magic method. Let me check this way. Could be that it will get a lot easier.

Reply:
Nice!

Reply:
I have removed the coupling between the property class and the DTO instance. This way we don't have to clone the properties and only cache them. The DTO instance is now passed to the `set()` method. I've also marked the property class as `@internal` to prevent the user from using it.
https://github.com/spatie/data-transfer-object/pull/66/commits/30566493cde2672f999af5209d536ade3f5e8705